crates/oxc_semantic/src/builder.rs

```diff
@@ -35,6 +35,18 @@ struct UnusedLabels<'a> {
     labels: Vec<AstNodeId>,
 }
 
+#[derive(Debug, Clone)]
+pub struct VariableInfo {
+    pub name: Atom,
+    pub span: Span,
+    pub scope_id: ScopeId,
```

Comment:
If you add the `symbol_id` here, then it's a single loop over `RedeclareVariables` 🤔 

`scope_id` seems to be unused.

Reply:
Yeah, but it's not enough loop over `RedeclareVariables` enough, because such as `globalThis` these globals are not stored in it.